lib/ssh_scan/config_loader/yaml.rb

```diff
@@ -0,0 +1,53 @@
+require 'yaml'
+
+module SSHScan
+  module ConfigLoader
+    class Yaml
+      def initialize(config, class_name = nil, mode = 'file', logger = nil)
```

Comment:
I think we should drop logging altogether, if we run into an exception in config loading, it should be a hard failure.  I think this simplifies and sort eliminates this as an issue from a "what do we pass into this object to get logging?" problem